> /__w/1/s/src/Framework/ref/Microsoft.AspNetCore.App.Ref.csproj : error NU1605: Detected package downgrade: System.IO.Pipelines from 4.7.1 to 4.7.0. Reference the package directly from the project to select a different version.  [/__t/dotnet/sdk/3.1.100/NuGet.targets]

I'll add NU1605 to the ref pack's NoWarn, but will we wind up transitively bringing in the higher version anyway? CC @dougbu @JunTaoLuo

--- a/eng/SharedFramework.External.props
+++ b/eng/SharedFramework.External.props
@@ -7,7 +7,7 @@
 <Project>
 
   <!-- For the targeting pack, always use packages with PatchVersion=0 -->
-  <PropertyGroup Condition="'$(MSBuildProjectName)' == 'Microsoft.AspNetCore.App.Ref' OR $(IsReferenceAssemblyProject) ">
+  <PropertyGroup Condition="'$(MSBuildProjectName)' == 'Microsoft.AspNetCore.App.Ref' OR '$(IsReferenceAssemblyProject)' == 'true' ">
     <SystemIOPipelinesPackageVersion>$(SystemIOPipelinesPackageVersion.Split('.')[0]).$(SystemIOPipelinesPackageVersion.Split('.')[1]).0</SystemIOPipelinesPackageVersion>
     <SystemSecurityCryptographyXmlPackageVersion>$(SystemSecurityCryptographyXmlPackageVersion.Split('.')[0]).$(SystemSecurityCryptographyXmlPackageVersion.Split('.')[1]).0</SystemSecurityCryptographyXmlPackageVersion>
     <MicrosoftWin32SystemEventsPackageVersion>$(MicrosoftWin32SystemEventsPackageVersion.Split('.')[0]).$(MicrosoftWin32SystemEventsPackageVersion.Split('.')[1]).0</MicrosoftWin32SystemEventsPackageVersion>

--- a/eng/SharedFramework.External.props
+++ b/eng/SharedFramework.External.props
@@ -7,7 +7,7 @@
 <Project>
 
   <!-- For the targeting pack, always use packages with PatchVersion=0 -->
-  <PropertyGroup Condition="'$(MSBuildProjectName)' == 'Microsoft.AspNetCore.App.Ref'">
+  <PropertyGroup Condition="'$(MSBuildProjectName)' == 'Microsoft.AspNetCore.App.Ref' OR $(IsReferenceAssemblyProject) ">
     <SystemIOPipelinesPackageVersion>$(SystemIOPipelinesPackageVersion.Split('.')[0]).$(SystemIOPipelinesPackageVersion.Split('.')[1]).0</SystemIOPipelinesPackageVersion>
     <SystemSecurityCryptographyXmlPackageVersion>$(SystemSecurityCryptographyXmlPackageVersion.Split('.')[0]).$(SystemSecurityCryptographyXmlPackageVersion.Split('.')[1]).0</SystemSecurityCryptographyXmlPackageVersion>
     <MicrosoftWin32SystemEventsPackageVersion>$(MicrosoftWin32SystemEventsPackageVersion.Split('.')[0]).$(MicrosoftWin32SystemEventsPackageVersion.Split('.')[1]).0</MicrosoftWin32SystemEventsPackageVersion>

--- a/src/Framework/ref/Microsoft.AspNetCore.App.Ref.csproj
+++ b/src/Framework/ref/Microsoft.AspNetCore.App.Ref.csproj
@@ -9,7 +9,8 @@
     <PackageId>$(TargetingPackName)</PackageId>
     <VersionPrefix>$(TargetingPackVersionPrefix)</VersionPrefix>
     <!-- This is a reference package and does not include any dependencies. -->
-    <NoWarn>$(NoWarn);NU5128</NoWarn>
+    <!-- We also disable warnings about package downgrades, because we intentionally reference lower versions of certain packages -->
+    <NoWarn>$(NoWarn);NU5128;NU1605</NoWarn>
 
     <PackageDescription>Provides a default set of APIs for building an ASP.NET Core application. Contains reference assemblies, documentation, and other design-time assets.
 

